file/http.lisp

```diff
@@ -199,17 +199,23 @@
     (read-http-response con)))
 
 (defun make-http-stream (pathname body element-type external-format)
-  (if (mezzano.internals::type-equal element-type 'character)
-      (make-instance 'http-character-stream
-                     :path pathname
-                     :direction :input
-                     :buffer body
-                     :length (length body))
-      (make-instance 'http-binary-stream
-                     :path pathname
-                     :direction :input
-                     :buffer body
-                     :length (length body))))
+  (let ((cache (make-hash-table :weakness :key-and-value))
+        (body-length (length body)))
+    (setf (gethash 0 cache) body)
```

Comment:
This isn't going to work. The body may be lost immediately after make-http-stream returns as there are no external references to it. There's no way for the http stream to recover it either as the contents are downloaded at open time.
file-cache doesn't seem like a good fit for http stream as it is not block-oriented.
This also touches implementation details of file-cache in a dubious way. Changes to how the cache is maintained will require changes here.